java/arcs/core/entity/CollectionHandle.kt

```diff
@@ -142,4 +137,24 @@ class CollectionHandle<T : Storable, R : Referencable>(
     }.filterNotTo(mutableSetOf()) {
         storageAdapter.isExpired(it)
     }
+
+    /** Configuration required to instantiate a [CollectionHandle]. */
+    class Config<T : Storable, R : Referencable>(
+        /** See [BaseHandleConfig.name]. */
+        name: String,
+        /** See [BaseHandleConfig.spec]. */
+        spec: HandleSpec<out Entity>,
+        /**
+         * Interface to storage for [RawEntity] objects backing an `entity: T`.
+         *
+         * See [BaseHandleConfig.storageProxy].
+         */
+        val proxy: CollectionProxy<R>,
+        /** Will ensure that necessary fields are present on the [RawEntity] before storage. */
+        val storageAdapter: StorageAdapter<T, R>,
+        /** See [BaseHandleConfig.dereferencerFactory]. */
+        dereferencerFactory: EntityDereferencerFactory,
+        /** See [BaseHandleConfig.particleId]. */
+        particleId: String
+    ) : BaseHandleConfig(name, spec, proxy, dereferencerFactory, particleId)
```

Comment:
Is there any other way of doing this where you don't have to repeat so many of the same arguments?